explorer/interpreter/type_checker.cpp

```diff
@@ -392,6 +392,56 @@ static auto IsConcreteType(Nonnull<const Value*> value) -> bool {
   return IsType(value) && !TypeContainsAuto(value);
 }
 
+// Returns whether the given value is template-dependent, that is, if it
+// depends on any template paramaeter.
+static auto IsTemplateDependent(Nonnull<const Value*> value) -> bool {
+  // A VariableType is template dependent if it names a template binding.
+  if (auto* var_type = dyn_cast<VariableType>(value)) {
+    return var_type->binding().binding_kind() ==
+           GenericBinding::BindingKind::Template;
+  }
+
+  static constexpr auto is_dependent_value = [](auto&& x) {
```

Comment:
```suggestion
  static constexpr bool is_dependent_value = [](auto&& x) {
```

WDYT of specifying the type, since the deduction is indirect?

Reply:
The type of `is_dependent_value` is a lambda closure type, not `bool`. There's no way to write it directly.

Reply:
We could do this if it would help:
```suggestion
  static constexpr auto is_dependent_value = [](auto&& x) -> bool {
```

Reply:
My mistake, but yes seeing types helps me. :)